app/lib/rocketchat.js

```diff
@@ -934,12 +934,13 @@ const RocketChat = {
 	onStreamData(...args) {
 		return this.sdk.onStreamData(...args);
 	},
-	emitTyping(room, typing = true) {
+	emitUserActivity(room, typing = true, activityType, extras = {}) {
 		const { login, settings } = reduxStore.getState();
-		const { UI_Use_Real_Name } = settings;
+		const { UI_Use_Real_Name, Use_New_Activity_Indicators } = settings;
 		const { user } = login;
 		const name = UI_Use_Real_Name ? user.name : user.username;
-		return this.methodCall('stream-notify-room', `${ room }/typing`, name, typing);
+		const stream = Use_New_Activity_Indicators ? 'user-activity' : 'typing';
```

Comment:
Should emit both if you accept my comment on the other PR

Reply:
Maybe we can make the backend control this and emit the old event as well if the setting is enabled. Something to check

Reply:
Here you need to check the server version and only emit the new event when the connected server is the version containing this improvement. We still don't know when it will be released but you can keep a specific number and make a note in the PR's description to update it when we have the right info.